cypress/integration/page.spec.js

```diff
@@ -1,7 +1,110 @@
-// WRITE TESTS HERE
+/// <reference types="cypress" />
 
 describe('Page', () => {
-  it('should be visitable', () => {
+  beforeEach(() => {
     cy.visit('/');
   });
+
+  it('user details should be updated after selecting a different user', () => {
+    cy.contains('5')
+      .click()
+
+    cy.contains('Chelsey Dietrich')
+      .should('have.text', 'Chelsey Dietrich')
+
+    cy.contains('6')
+      .click()
+
+    cy.contains('Mrs. Dennis Schulist')
+      .should('have.text', 'Mrs. Dennis Schulist')
+  });
+
+  it('no request to the server after selecting the same user again', () => {
+    cy.intercept('https://mate.academy/students-api/users/5', cy.spy().as('apiCall'))
+
+    cy.contains('5')
+      .click()
+      .click()
+
+    cy.get('@apiCall')
+      .its('callCount')
+      .should('equal', 1)
+  });
+
+  it('selected user is cleared after clicking "Clear" button', () => {
+    cy.contains('5')
+      .click()
+
+    cy.contains('Chelsey Dietrich')
+      .should('have.text', 'Chelsey Dietrich')
+
+    cy.contains('Clear')
+      .click()
+
+    cy.contains('No user selected')
+      .should('have.text', 'No user selected')
+  });
+
+  it('todos can be filtered by title', () => {
+    cy.request('https://mate.academy/students-api/todos')
```

Comment:
ok, but I suggest using mocks instead. There can be 0 items

Reply:
fixed